### ISSUE TITLE
Difference in estimated difference due to difference in rules

### ISSUE THREAD

**Issue by hope366** [lightvector/KataGo#515]
See the screenshot below. This is the final phase of the title match that took place in Japan the other day. I set it to the Japanese rule, Komi 6.5 and analyzed it.
![japanese](https://user-images.githubusercontent.com/63999713/124298618-6ec6d080-db97-11eb-913f-f989360b90ac.png)
Next, I analyzed the exact same situation with the Chinese rule, Komi 7.5.
![chinese](https://user-images.githubusercontent.com/63999713/124298848-b4839900-db97-11eb-937e-c3255c73a9d4.png)
There is a slight difference in the estimated difference. Why is this? The Japanese rule and the Chinese rule differ in how to count the seki, which may cause a difference in the result, but I think that both are the same in this case.
The case where the result differs depending on how to count the seki is, for example, the case shown in the figure below.
![seki](https://user-images.githubusercontent.com/63999713/124302122-b3546b00-db9b-11eb-8cd8-236d5064d78d.png)
 In this form, Black finally forces White to pass, and only you can hit it twice in a row. So, under the Chinese rules, black has a 2 point advantage.
However, since Seki in the phase diagram of this title battle is not of such a nature, there should be no difference in superiority or inferiority depending on the rules.
So why does the difference in the rules make a difference in the estimated difference? Is it because KataGo's strategy and tactics differ depending on the rules?
I added stones from the phase diagram presented above and analyzed it with no place to put stones anymore, but this time the estimated difference due to the difference in rules was only 0.2 points. ..
I think the reason why the estimated difference is so small is that there is no place to demonstrate strategy and tactics.

We apologize for the fact that the English may have been difficult to understand because it uses the translation function.

**Comment by lightvector** [comment]
Isn't the lower right seki genuinely scored differently under the two rules? Black gets a point for the false eye in Chinese rules. 

See a simpler example here. In Japanese rules, the territory score is 0. In Chinese rules, Both black and white have played equal number of stones (14 stones each), so the stone value cancels out, but black has 2 territory points while white has only 1, so white has +1. So the rules make a difference for this seki, it is not the same.
![image](https://user-images.githubusercontent.com/11942395/124306168-07c40000-db34-11eb-9a07-da68f1db35dc.png)

The second thing is that you cannot consider only one way of playing out the position, because the difference between Japanese and Chinese will also vary not just because of the seki, but they will also differ between who gets the last move on the board. If you add stones to play out the board, you will only get one possible result, but you will miss the other. So you have to consider at least 2 different ways of adding stones to the board, one for each possible parity, to make a valid analysis.

Have you accounted for both of these factors? It doesn't sound to me like you have.

Lastly, KataGo sometimes cares less about estimating the score accurately when difference is large, and 10 points is a huge difference at professional level. You will get more accurate results (as well as a better sense of the uncertainty remaining) if you manually adjust komi and see what komi makes the winrate closest to 50%, and how the winrate in general varies as a function of the komi around there.

**Comment by lightvector** [comment]
I studied it a bit more and I think KataGo is correct here. The presence of this seki shape makes Chinese rules a lot better for black than Japanese rules.

## Odd dame parity

Let's consider this endgame, where komi is 0, and there are no captured stones so far.
![image](https://user-images.githubusercontent.com/11942395/124312286-0a772300-db3d-11eb-8d0b-7e0735f50726.png)

Both sides have played 22 stones on the board, therefore it must be black's turn next. The endgame proceeds like shown here. Black fills the final dame on the board (move 3) aside from the seki, white passes (move 4), and black captures the seki stone (move 5), and the game ends.
![image](https://user-images.githubusercontent.com/11942395/124313110-6a21fe00-db3e-11eb-96be-08683fdcc320.png)

Under Japanese rules, black wins by 1 point. Under Chinese rules, black wins by 4 points. Therefore, given equal komi, Chinese rules in this situation are 3 points better.

## Even dame parity

As I said in my previous post, it is necessary to consider the opposite parity as well, where the dame are even instead of odd, so that white is the one who fills the last dame. So we try this board position:

![image](https://user-images.githubusercontent.com/11942395/124312815-f08a1000-db3d-11eb-8a52-9e533b2bf1c8.png)

Both sides have played 23 stones on the board, so it must be black's turn next. The endgame proceeds like shown here. This time the dame are even, so white indeed gets the last dame (move 4), then black captures the seki stone (move 5), and the game ends.
![image](https://user-images.githubusercontent.com/11942395/124313165-7a39dd80-db3e-11eb-862d-4a6c48bdcf88.png)

Under Japanese rules, black wins by 2 points. Under Chinese rules, black wins by 4 points. Therefore, given equal komi, Chinese rules in this situation are 2 points better for black.

So depending on the even/odd dame parity, Chinese rules are either 3 points better for black, or 2 points better for black when this kind of seki shape exists on the board. So the difference on average is 2.5 points.

If we use a higher komi for white in Chinese rules (7.5 instead of 6.5), then the advantage to black for Chinese rules versus Japanese rules drops from 2.5 points to 1.5 points. And that is extremely close to the score difference displayed by KataGo in your screenshot.

**Comment by lightvector** [comment]
Followup: given equal komi, normally Chinese rules are on average 0.5 points for better for black than Japanese rules, since black has a 50% chance of getting an extra dame point, which counts as an extra point in Chinese rules.

When this seki is present, how do the exact same move sequences become 2.5 points? Where are the two extra points from?

* One point is due to being able to count G1 as a territory point. This is territory in Chinese rules but not in Japanese rules.
* The other point is due to the fact that black gets one extra stone (move 5) *after* all the dame are filled.

**Comment by hope366** [comment]
First of all, thank you for your quick answer.

I see, I should add 0.5 bonus points for black to the difference in the number of positions in the seki.

Normally, the Japanese rule is Komi 6.5, and the Chinese rule is Komi 7.5, so the number calculated from the above numbers to give Black a 1-point disadvantage is the difference between the numbers that KataGo shows as the estimated point difference in both rules.

In other words, if seki exists, the estimated point difference between Japanese and Chinese rules would be 0.5, 1.5, 2.5 ... etc. Is this correct?

**Comment by lightvector** [comment]
Yes, the average point difference between Japanese and Chinese rules will usually end in ".5" because the difference between the two depends on the 50% chance of who gets the last dame, which even for AI is very difficult to predict, so it is basically random. The exception is when it becomes possible to read all the way to the end of the game or otherwise solve the position, and actually read who will get the last dame.

In normal positions, on average, Japanese rules komi 6.5 ~= Chinese rules komi 7.0, so if you are using Chinese rules komi 7.5, white has an 0.5 extra advantage compared to Japanese rules komi 6.5.

Your seki gives 2 more points to black under Chinese rules. So then with such a seki, Chinese rules 7.5 komi instead becomes 1.5 points better for black compared to Japanese rules komi 6.5.

**Comment by hope366** [comment]
>In normal positions, on average, Japanese rules komi 6.5 ~= Chinese rules komi 7.0, so if you are using Chinese rules komi 7.5, white has an 0.5 extra advantage compared to Japanese rules komi 6.5.

The reason is that compared to the Japanese rule, the Chinese rule may have an average 0.5 point advantage over black due to the right to fill in the last dame.
Does this explanation match?

Next, in order to realize what I learned this time, I considered the difference in the figure presented at the beginning, that is, the reason why the estimated point difference of Katago differs by about 1.5 points between Japan and China.
![無題](https://user-images.githubusercontent.com/63999713/124348292-1853a380-dc24-11eb-8bb1-cbd6ff699732.png)
Let's pay attention to the lower right part.
The number of stones used is 10 for black and 7 for white, but since they have not passed each other, white should be placed elsewhere because there are 3 more black stones, so stones You shouldn't have to consider the difference due to the number.
As for the remaining part, at the end Black can force White to pass and place stones twice in a row. In other words, 2 points black is advantageous. Add this and 0.5 bonus points for a total of 2.5 points. Both Japanese rules have 0 points, so the Chinese rules have a 2.5 point advantage over black compared to the Japanese rules. However, in reality, the Chinese rule is more disadvantageous to Komi by 1.0 points, so the difference between the leads of the deducted black is 1.5 points.
The difference between the numbers shown by KataGo on the board is 1.4 points, but they are almost the same.
Is this way of thinking correct?

**Comment by hope366** [comment]
Let me ask you another question.
The first presented phase diagram is displayed again.
![ss](https://user-images.githubusercontent.com/63999713/124356371-b198af00-dc50-11eb-867b-f1c947a41fae.png)
Regarding the lower right part of the seki, there are white horikomi and black tori in advance at 17-19. In this state, how many points of black exist in the lower right part of the seki? Some people have answered the question as follows. Is this correct?

"There are 2 points of black under Japanese rules"
"There are 4 or 5 points of black under the Chinese rule."

**Comment by lightvector** [comment]
> Is this way of thinking correct?
Yes, this is correct.

> "There are 2 points of black under Japanese rules"
> "There are 4 or 5 points of black under the Chinese rule."

This is a completely way of thinking about this position. But I would count it as 1 and 3, instead of 2 and 4.

There is only 1 point for black under the Japanese rules, because nobody has territory there, black only gets to capture the stone there later, which is 1 point. 
And there are effectively 3 net points for black under Chinese rules, because black counts all 3 points of territory plus 1 more because white's stone is dead (which is one less point for white because white will have one less living stone somewhere else on the board), but white also has 1 point of territory too. So 3+1 - 1 = 3.

This is a different way of thinking, but it's equally correct as the other way of thinking.

**Comment by hope366** [comment]
I have forgotten to explain.
Locations 17-19 are empty, but early on there are white horikomi and black captures.
There are white stones captured in 17-19 and 14-19 white stones that can be captured later, so there is an advantage of 2 points on the black side according to Japanese rules.
Next, let's consider the Chinese rules.
Number of living black stones = 10 Black territory = 3.5 (including half of common dame)
Number of living white stones = 7 White territory = 1.5 (including half of common dame)
13.5-8.5 = 5
If you think more carefully here, you can see that there is a difference in the number of stones used in the seki part. Black has 13 on the board, which is fine, but white has 7 on the board, 1 captured at 17-19 and 1 dead at 14-17, for a total of 9 is.
In other words, white uses one less stone than black, so it should be able to be placed somewhere else.
In other words, black has an advantage of 5 points when considering only the part of the seki, but when considering the entire board, the advantage of black due to the presence of seki is 4 points.
Under the Japanese rule, black had an advantage of 2 points due to the existence of Seki, but under the Chinese rule, black had an advantage of 4 points.
Therefore, when considering the board as a whole, the advantage of black due to the difference in rules is 2 points, and the total rule is added by adding -1 point due to the difference in Komi and +0.5 points regarding the right to fill the last dame. The advantage of black due to the difference is +1.5 points. The calculation result of KataGo is also about 1.5 points difference.

'm sorry to ask the same question again, but is this idea correct?

**Comment by lightvector** [comment]
Yes, I think that is correct. Good analysis. :)

**Comment by hope366** [comment]
Thanks to your polite explanation I was able to understand well.
It's fun to think about one thing thoroughly 😄
thank you very much.
